### ISSUE TITLE
Call to a member function getGroups() on null within logout

### ISSUE THREAD

**Issue by jozefrebjak** [codeigniter4/shield#214]
I've got an Error within logout.

![CleanShot 2022-06-07 at 21 33 27@2x](https://user-images.githubusercontent.com/36922215/172467196-3ccf6a17-64f8-4cf2-80f9-4a0a9271f84a.png)

**Comment by jozefrebjak** [comment]
I saw that there was a fix for something related with toolbar, but I had this issue today with fresh install of CI 4.2 and latest shield and PR #211 was merged two days ago.

**Comment by kenjis** [comment]
How did you install Shield?

**Comment by jlopes90** [comment]
The same error continues. I registered an account after logging in and it doesn't work to log out. So I did migrate:refresh then got this error.

**Comment by jozefrebjak** [comment]
@kenjis installed as docs describe instalation, but it seems we need to specify develop version to use, becase beta version was installed.

**Comment by kenjis** [comment]
> I registered an account after logging in and it doesn't work to log out. So I did migrate:refresh then got this error.

It seems you deleted the user's group data in the table.

**Comment by jlopes90** [comment]
> > I registered an account after logging in and it doesn't work to log out. So I did migrate:refresh then got this error.
> 
> It seems you deleted the user's data in the table.

Well, now I'm stuck with this error. It's not to resolve the users table or others. It has to be corrected, so it is possible to login or register. I think myth-auth works because it checks "users" in the database.

**Comment by kenjis** [comment]
> It seems you deleted the user's data in the table.

But if there is no user, nobody can't logged in. So the toolbar shows not logged in.

@jlopes90 Can you show us the output of the following commands?

```
$ php spark db:table user
$ php spark db:table auth_groups_users
```

**Comment by jlopes90** [comment]
> > It seems you deleted the user's data in the table.
> 
> But if there is no user, nobody can't logged in. So the toolbar shows not logged in.
> 
> @jlopes90 Can you show us the output of the following commands?
> 
> ```
> $ php spark db:table user
> $ php spark db:table auth_groups_users
> ```

![image](https://user-images.githubusercontent.com/50026015/172717408-97b387cb-8bb6-4275-b63b-2b5adf53b817.png)

![image](https://user-images.githubusercontent.com/50026015/172717466-57352b7d-c75a-4cae-b2b8-790e18227f1f.png)

All are empty, because I did `migrate:refresh`.

I already tried to enter the address for example: http://localhost:8082/login, still the same error.

**Comment by jlopes90** [comment]
> @jlopes90 May I ask you to check on another browser for performance? And announce the result.

apparently, I've already tested other browsers (firefox or edge) and it works normal, login appears

**Comment by datamweb** [comment]
Thanks, please send photos of all error tabs in Chrome.

**Comment by jlopes90** [comment]
it's the same error https://github.com/codeigniter4/shield/issues/214#issue-1263774736

**Comment by datamweb** [comment]
Yes I know but we need to check some items, please choose tabs(server, req....) and take photos

**Comment by jlopes90** [comment]
https://user-images.githubusercontent.com/50026015/172719909-dcf32b43-8754-42f3-9193-6199cbde6fc6.jpg

save, change .jpg to .html and drag to browser

**Comment by kenjis** [comment]
@jlopes90 Remove the cookies (`ci_session` and `remember`) in your Chrome.

**Comment by datamweb** [comment]
![IMG_20220609_020213.png](https://user-images.githubusercontent.com/9530214/172721222-20b8d401-f800-428e-9ad9-085651b47ebf.png)

We need the **code** in each tabs, you may send the **code** ?

**Comment by jlopes90** [comment]
> Yes I know but we need to check some items, please choose tabs(server, req....) and take photos
>
> ![IMG_20220609_020213.png]
> 
> We need the **code** in each **tabs**, you may send the **code** ?

https://github.com/codeigniter4/shield/issues/214#issuecomment-1150430869

**Comment by jlopes90** [comment]
> @jlopes90 Remove the cookies (`ci_session` and `remember`) in your Chrome.

already removed cookies (`ci_session` and `remember`) and works as normal.

**Comment by datamweb** [comment]
@kenjis , I could not see the content with my mobile phone, did you see https://github.com/codeigniter4/shield/issues/214#issuecomment-1150430869?

**Comment by jlopes90** [comment]
> @kenjis , I could not see the content with my mobile phone, did you see https://github.com/codeigniter4/shield/issues/214#issuecomment-1150430869?

Also gives mobile, save, open app file with download folder, change .jpg to .html and open

**Comment by kenjis** [comment]
@datamweb Yes.

**Comment by datamweb** [comment]
@jlopes90 , we wanted to see exactly which cookie was causing this problem. 
**Thank you**.

**Comment by kenjis** [comment]
Okay, I got the cause.
I will fix this.

**Comment by datamweb** [comment]
Thanks @kenjis , forgive me for answering in some cases before the members.
My goal is to help the group.
I hope it is not rude.

**Comment by kenjis** [comment]
I sent a PR to fix:  #222

@datamweb No problem! There is no rule that non member should not answer.

**Comment by kenjis** [comment]
Fixed in the latest `develop`.
